ee/packages/omnichannel-services/src/OmnichannelTranscript.ts

```diff
@@ -232,8 +232,20 @@ export class OmnichannelTranscript extends ServiceClass implements IOmnichannelT
 					continue;
 				}
 
-				const fileBuffer = await uploadService.getFileBuffer({ file: uploadedFile });
-				files.push({ name: file.name, buffer: fileBuffer, extension: uploadedFile.extension });
+				try {
+					const fileBuffer = await uploadService.getFileBuffer({ file: uploadedFile });
+					files.push({ name: file.name, buffer: fileBuffer, extension: uploadedFile.extension });
+				} catch (e: unknown) {
+					this.log.error(`Failed to get file ${file._id}`, e);
+					// Push empty buffer so parser processes this as "unsupported file"
+					files.push({ name: file.name, buffer: null });
+
+					if ((e as Error).message === 'MAX_PAYLOAD_EXCEEDED') {
+						this.log.error(
+							`File is too big to be processed by NATS. See NATS config for allowing bigger messages to be sent between services`,
+						);
+					}
```

Comment:
should we just increase or have some sort of file-split mechanism to work around it? like send 2MB-2MB-2MB -> 6MB file?

Reply:
ps: can be done later, not a blocker for this pr imo.

Reply:
That would be cool, but yeah, for now, this is what we have.

Would be nice to send some streams over nats or maybe allowing the services to get a buffer directly with some package, idk.

(Or just chunked buffers, that works too)

Reply:
moleculer supports Streams so in theory we could use them to transfer bigger files.. so after receiving the whole stream we could create a buffer from it again.. even better, we could change our APIs to support streams as well

Reply:
That would be pretty cool! I can spend some time investigating that, since our current way of sending the whole buffer seems a bit overkill. We just do that with images but yeah.